### ISSUE TITLE
BUG APIGEE Google Native SDK 

### ISSUE THREAD

**Issue by LuisAyazo** [pulumi/pulumi-google-native#429]
## Hello!

- Vote on this issue by adding a 👍 reaction
- To contribute a fix for this issue, leave a comment (and link to your pull request, if you've opened one already)

## Issue details
Trying to create a Api Proxy from Google Native SDK, i am getting the following error despite all the atribute are passed especially  'multipart/form-data.

ERROR:
error: error sending request: googleapi: Error 400: proxy must be encoded as 'application/octet-stream' or 'multipart/form-data'

Api Proxy DOC:
https://www.pulumi.com/registry/packages/google-native/api-docs/apigee/v1/api/

### Steps to reproduce

1. pulumi componente resource:
```
# -*- coding: utf-8 -*-
from typing import Mapping, Optional, Sequence

from pulumi import ComponentResource, ResourceOptions, log

# use google classic because issue https://github.com/pulumi/pulumi-google-native/issues/292
from pulumi_google_native.apigee import v1 as gcp

# import pulumi_gcp as gcp
from pydantic import BaseModel

__all__ = ["ApiProxy", "ApiProxyArgs"]


class ApiProxyArgs(BaseModel):
    """
    The arguments necessary to construct a 'Apigee Api Proxy' resource.
    """

    # required
    organization_id: str

    # optional
    action: Optional[str] = None
    content_type: Optional[str] = None
    data: Optional[str] = None
    extensions: Sequence[Mapping[str, str]] = {}
    name: Optional[str] = None
    # validate: Optional[str] = None

    class Config:
        arbitrary_types_allowed = True


class ApiProxy(ComponentResource):
    def __init__(
        self,
        resource_name: str,
        args: ApiProxyArgs,
        opts: Optional[ResourceOptions] = None,
    ):

        super().__init__(
            t="devops-plm-gcp-infra:apigeeApiProxy/alpha:apigeeApiProxy",
            name=f"{resource_name}-Apigee-Api-Proxy",
            props={},
            opts=opts,
        )
        self_parent = self
        child_opts = ResourceOptions(parent=self_parent)
        try:
            gcp.Api(
                resource_name=resource_name,
                organization_id=args.organization_id,
                name=args.name,
                data=args.data,
                content_type="application/octet-stream",
                extensions=args.extensions,
                action=args.action,
                opts=child_opts,
            )
        except Exception as ex:
            log.error(f"Apigee Api Proxy -> {ex.args[0]}")
```

2. call the componente and execute pulumi up. 
3. we are getting this error: **error sending request: googleapi: Error 400: proxy must be encoded as 'application/octet-stream' or 'multipart/form-data'** 
4.

Expected: 
I expect the pulumi up command be able to create the api proxy by uploading the zip file
Actual: 
we are getting this error: **error sending request: googleapi: Error 400: proxy must be encoded as 'application/octet-stream' or 'multipart/form-data'**

**Comment by viveklak** [comment]
According to the discovery docs, the content-type needs to be set to  'application/octet-stream' or 'multipart/form-data': https://github.com/pulumi/pulumi-google-native/blob/master/discovery/apigee_v1.json#L5459

**Comment by LuisAyazo** [comment]
we are setting those values and still it does not work and throws the same error.

**Comment by viveklak** [comment]
A fix is available in #590. We will look to get it merged next week.
